src/cgroup.c

```diff
@@ -279,19 +279,35 @@ gboolean check_cgroup2_oom()
 	ssize_t read;
 	while ((read = getline(&line, &len, fp)) != -1) {
 		long int counter;
+		int memcmp_oom, memcmp_oom_kill, len;
+		const int oom_len = 4, oom_kill_len = 9;
 
-		if (read < 6 || memcmp(line, "oom ", 4))
+		if (read < oom_len + 2 && read < oom_kill_len + 2)
```

Comment:
wouldn't we skip to check `oom `, if the len is `<= 11`?

I think we need to change the check to be something like:

```
if (read >= oom_len + 2)
    memcmp_oom = memcmp(line, "oom ", oom_len);
if (read >= oom_kill_len + 2)
    memcmp_oom = memcmp(line, "oom_kill ", oom_len);
```

instead of the `continue` block.

Reply:
Yeah I guess we can merge the `memcmp` with the `read` len checks. Updated the paths a bit to

```c
		if (read >= oom_len + 2 && memcmp(line, "oom ", oom_len) == 0)
			len = oom_len;
		else if (read >= oom_kill_len + 2 && memcmp(line, "oom_kill ", oom_kill_len) == 0)
			len = oom_kill_len;
		else
			continue;
```

Thank you for the hint!